sound/soc/sof/debug.c

```diff
@@ -640,7 +640,7 @@ static int memory_info_update(struct snd_sof_dev *sdev, char *buf, size_t buff_s
 	int ret;
 	int i;
 
-	reply = kmalloc(SOF_IPC_MSG_MAX_SIZE, GFP_KERNEL);
+	reply = kmalloc(sdev->host_box.size, GFP_KERNEL);
```

Comment:
do we have anywhere in the code something that checks that this size is a reasonable value, and that there is no mismatch.
I may be paranoid here, but can we trust blindly the firmware here?

Reply:
we have a simple check https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/loader.c#L468-L471. but we don't check its range. what is the reasonable range ?  @keyonjie

Reply:
add check, thanks!